spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,236 @@
+            .toArray(SortOrder[]::new);
+
+    return Pair.of(Distributions.hash(bucketNum, bucketColumns), sortOrders);
+  }
+
+  private static org.apache.spark.sql.connector.expressions.Transform toSparkBucketTransform(
+      Distribution distribution, SortOrder[] sortOrders) {
+    if (distribution == null) {
+      return null;
+    }
+
+    switch (distribution.strategy()) {
+      case NONE:
+        return null;
+      case HASH:
+        int bucketNum = distribution.number();
+        String[] bucketFields =
+            Arrays.stream(distribution.expressions())
+                .map(
+                    expression ->
+                        getFieldNameFromGravitinoNamedReference((NamedReference) expression))
+                .toArray(String[]::new);
+        if (sortOrders == null || sortOrders.length == 0) {
+          return Expressions.bucket(bucketNum, bucketFields);
+        } else {
+          String[] sortOrderFields =
+              Arrays.stream(sortOrders)
+                  .map(
+                      sortOrder ->
+                          getFieldNameFromGravitinoNamedReference(
+                              (NamedReference) sortOrder.expression()))
+                  .toArray(String[]::new);
+          return createSortBucketTransform(bucketNum, bucketFields, sortOrderFields);
+        }
+        // Spark doesn't support EVEN or RANGE distribution
+      default:
+        throw new NotSupportedException(
+            "Doesn't support distribution strategy: " + distribution.strategy());
+    }
+  }
+
+  private static Expression[] transToGravitinoNamedReference(
+      List<org.apache.spark.sql.connector.expressions.NamedReference> sparkNamedReferences) {
+    return sparkNamedReferences.stream()
+        .map(sparkReference -> NamedReference.field(sparkReference.fieldNames()))
+        .toArray(Expression[]::new);
+  }
+
+  public static org.apache.spark.sql.connector.expressions.Transform createSortBucketTransform(
+      int bucketNum, String[] bucketFields, String[] sortFields) {
+    return LogicalExpressions.bucket(
+        bucketNum, createSparkNamedReference(bucketFields), createSparkNamedReference(sortFields));
+  }
+
+  // columnName could be "a" or "a.b" for nested column
+  public static IdentityTransform createSparkIdentityTransform(String columnName) {
+    return IdentityTransform.apply(Expressions.column(columnName));
+  }
+
+  private static org.apache.spark.sql.connector.expressions.NamedReference[]
+      createSparkNamedReference(String[] fields) {
+    return Arrays.stream(fields)
+        .map(Expressions::column)
+        .toArray(org.apache.spark.sql.connector.expressions.NamedReference[]::new);
+  }
+
+  // Gravitino use ["a","b"] for nested fields while Spark use "a.b";
+  private static String getFieldNameFromGravitinoNamedReference(
+      NamedReference gravitinoNamedReference) {
+    return String.join(ConnectorConstants.DOT, gravitinoNamedReference.fieldName());
```

Comment:
@mchades Does Gravitino support nested fields? I remember `["a","b"]`, a means the table reference and b is the real column name?

Reply:
No, the array of fieldName is used to represent access nested fields. For example, the column `a` is struct type`{b int, c string}`, then we can use `a.b` or `a.c` to reference a  nested field